### PR TITLE
docs: profiler: Profiler docs

### PR DESCRIPTION
**Pull request by MarekPieta**
Change adds profiler conceptual documentation.

### CODE REVIEW

**Comment by MarekPieta** [comment]
@ru-fu, give me a moment please. I will add profiling custom event description (event which is not part of Event Manager)

File: include/profiler.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+Processing of the events is displayed as rectangles, visualizing the processing time.
+
+Use the **start/stop** button below the plot to pause or resume real time plot translation.
+Scroll to zoom in or out.
+When paused, scrolling zooms to the cursor location.
+When plotting in real time, scrolling zooms to the right edge of the plot.
+Use the middle mouse button to mark an event submission or processing for tracking, and to display the event data.
+
+When plotting is paused, you can click and drag with the left mouse button to pan the plot.
+Click the left and right mouse button simultaneously to place a horizontal line at the cursor location.
```
Comment (ru-fu) [inline include/profiler.rst]: I haven't seen the GUI, but vertical lines would be more common (if time is on the x axis)?

Comment (ru-fu) [reply include/profiler.rst]: Also, is it both buttons simultaneously? The input wasn't clear.

Comment (MarekPieta) [reply include/profiler.rst]: Yes, of course. I fixed it.
I was going to take a look at whole documentation after adding profiling custom events section and then ask you for review.

**Comment by MarekPieta** [comment]
@ru-fu , could you rereview?

**Comment by ru-fu** [comment]
Updated - looks good now. :)

**Comment by MarekPieta** [comment]
@ru-fu, I squashed commits. Could you take a look?

**Comment by MarekPieta** [comment]
@pdunaj, could you merge ?

**Pull request merged**